apps/web/src/views/universalFarms/hooks/useCakeEarning.ts

```diff
@@ -0,0 +1,50 @@
+import { formatBigInt } from '@pancakeswap/utils/formatBalance'
+import BigNumber from 'bignumber.js'
+import useAccountActiveChain from 'hooks/useAccountActiveChain'
+import { useCakePrice } from 'hooks/useCakePrice'
+import { useMemo } from 'react'
+import { useStakedPositionsByUser } from 'state/farmsV3/hooks'
+import { useAccountV2PendingCakeReward } from 'state/farmsV4/state/accountPositions/hooks/useAccountV2PendingCakeReward'
+import { getUniversalBCakeWrapperForPool } from 'state/farmsV4/state/poolApr/fetcher'
+import { PoolInfo } from 'state/farmsV4/state/type'
+
+export const useV2CakeEarning = (pool: PoolInfo | null | undefined) => {
+  const { account } = useAccountActiveChain()
+  const cakePrice = useCakePrice()
+  const { chainId, lpAddress } = pool || {}
+  const bCakeConfig = useMemo(() => {
+    return chainId && lpAddress
+      ? getUniversalBCakeWrapperForPool({ chainId, lpAddress, protocol: pool?.protocol })
+      : null
+  }, [chainId, lpAddress, pool?.protocol])
+  const { data: pendingCake } = useAccountV2PendingCakeReward(account, {
+    chainId,
+    lpAddress,
+    bCakeWrapperAddress: bCakeConfig?.bCakeWrapperAddress,
+  })
+  const earningsAmount = useMemo(() => +formatBigInt(BigInt(pendingCake ?? 0), 5), [pendingCake])
+  const earningsBusd = useMemo(() => {
+    return new BigNumber(earningsAmount ?? 0).times(cakePrice.toString()).toNumber()
+  }, [cakePrice, earningsAmount])
+
+  return {
+    earningsAmount,
+    earningsBusd,
+  }
+}
+
+export const useV3CakeEarning = (tokenIds: bigint[] = []) => {
```

Comment:
why the `useV3CakeEarning` not just use pool as arguments, keep same as the `useV2CakeEarning`

Reply:
the only needed arg is tokenIds, why we need the rest?